phoenix-core/src/main/java/org/apache/phoenix/coprocessor/GroupedAggregateRegionObserver.java

```diff
@@ -402,8 +405,8 @@ private RegionScanner scanUnordered(ObserverContext<RegionCoprocessorEnvironment
             }
 
             Region region = c.getEnvironment().getRegion();
-            region.startRegionOperation();
             try {
+                region.startRegionOperation();
```

Comment:
region.startRegionOperation() prevent from splitting and also if the region already closed we are throwing NSRE. I have moved into try block only to catch NSRE and throw Stale region boundary exception.